File: FMI/ANUL II/SEM 2/MPP/ProiectC#/Zboruri/Networking/JsonProtocolUtils.cs
using Model;

namespace Networking;

public class JsonProtocolUtils
{
    public static Response CreateOkResponse(){
        Response resp=new Response
        {
            Type = ResponseType.Ok
        };
        return resp;
    }
    
    public static Response CreateErrorResponse(String errorMessage){
        Response resp=new Response
        {
            Type = ResponseType.Error,
            ErrorMessage = errorMessage
        };
        return resp;
    }
    
    public static Request CreateLoginRequest(Angajat angajat)
    {
        var req = new Request
        {
            Type = RequestType.Login,
            Angajat = angajat
        };
        return req;
    }

    public static Request CreateLogoutRequest(Angajat angajat)
    {
        var req = new Request
        {
            Type = RequestType.Logout,
            Angajat = angajat
        };
        return req;
    }

    public static Response CreateGetAllFlightsResponse(List<Zbor> flights)
    {
        var resp = new Response
        {
            Type = ResponseType.GetFlights,
            Flights = flights
        };
        return resp;
    }

    public static Response CreateGetFlightsDestDateSeatsResponse(List<Zbor> flights)
    {
        var resp = new Response
        {
            Type = ResponseType.GetFlightsSeatsDestDate,
            Flights = flights
        };
        return resp;
    }

    public static Request CreateGetFlightsDestDateSeatsRequest(string dest, DateTime date, int min)
    {
        var req = new Request
        {
            Type = RequestType.GetFlightsSeatsDestDate,
            Destination = dest,
            Date = date,
            MinSeats = min
        };
        return req;
    }

    public static Request CreateGetAllFlightsRequest()
    {
        var req = new Request
        {
            Type = RequestType.GetAllFlights
        };
        return req;
    }

    public static Request CreateBuyTicketRequest(string client, string tara, string oras, List<TuristBilet> turisti, int nrLocuri, Zbor zbor)
    {
        var req = new Request
        {
            Type = RequestType.BuyTicket,
            Client = client,
            Tara = tara,
            Oras = oras,
            Turisti = turisti.Select(t => t.Turist).ToList(),
            NrLocuri = nrLocuri,
            Zbor = zbor
        };
        return req;
    }

    public static Response CreateBuyTicketResponse(Zbor oldZbor, Zbor zbor)
    {
        var resp = new Response
        {
            Type = ResponseType.BuyTicket,
            Zbor = zbor,
            OldZbor = oldZbor
        };
        return resp;
    }

    public static Request CreateFindZborRequest(int id)
    {
        var req = new Request
        {
            Type = RequestType.FindZbor,
            Id = id
        };
        return req;
    }

    public static Response CreateFindZborResponse(Zbor zbor)
    {
        var resp = new Response
        {
            Type = ResponseType.FindZbor,
            Found = zbor
        };
        return resp;
    }
}

File: FMI/ANUL II/SEM 2/MPP/ProiectJava/Zboruri/ZboruriNetworking/src/main/java/ro/mpp2024/network/jsonprotocol/JsonProtocolUtils.java
package ro.mpp2024.network.jsonprotocol;

import ro.mpp2024.domain.Zbor;
import ro.mpp2024.domain.Angajat;
import ro.mpp2024.utils.Constants;

import java.time.LocalDateTime;
import java.util.List;


public class JsonProtocolUtils {

    public static Response createOkResponse(){
        Response resp=new Response();
        resp.setType(ResponseType.Ok);
        return resp;
    }

    public static Response createErrorResponse(String errorMessage){
        Response resp=new Response();
        resp.setType(ResponseType.Error);
        resp.setErrorMessage(errorMessage);
        return resp;
    }

    public static Request createLoginRequest(Angajat angajat){
        Request req=new Request();
        req.setType(RequestType.Login);
        req.setAngajat(angajat);
        return req;
    }

    public static Request createLogoutRequest(Angajat angajat){
        Request req=new Request();
        req.setType(RequestType.Logout);
        req.setAngajat(angajat);
        return req;
    }

    public static Response createGetAllFlightsResponse(List<Zbor> flights) {
        Response resp=new Response();
        resp.setType(ResponseType.GetFlights);
        resp.setFlights(flights);
        return resp;
    }

    public static Response createGetFlightsDestDateSeatsResponse(List<Zbor> flights) {
        Response resp=new Response();
        resp.setType(ResponseType.GetFlightsSeatsDestDate);
        resp.setFlights(flights);
        return resp;
    }

    public static Request createGetFlightsDestDateSeatsRequest(String dest, LocalDateTime date, int min) {
        Request req=new Request();
        req.setType(RequestType.GetFlightsSeatsDestDate);
        req.setDestination(dest);
        req.setDate(date.format(Constants.DATE_TIME_FORMATTER));
        req.setMinSeats(min);
        return req;
    }

    public static Request createGetAllFlightsRequest() {
        Request req=new Request();
        req.setType(RequestType.GetAllFlights);
        return req;
    }

    public static Request createBuyTicketRequest(String client, String tara, String oras, List<String> turisti, int nrLocuri, Zbor zbor) {
        Request req=new Request();
        req.setType(RequestType.BuyTicket);
        req.setClient(client);
        req.setTara(tara);
        req.setOras(oras);
        req.setTuristi(turisti);
        req.setNrLocuri(nrLocuri);
        req.setZbor(zbor);
        return req;
    }

    public static Response createBuyTicketResponse(Zbor oldZbor, Zbor zbor) {
        Response resp=new Response();
        resp.setType(ResponseType.BuyTicket);
        resp.setZbor(zbor);
        resp.setOldZbor(oldZbor);
        return resp;
    }
}
